lib/layout/forceDirectedLayout.js

```diff
@@ -173,6 +172,36 @@ var springForce = function(edge) {
   }
 };
 
+var portLayoutHelper = function(ports, cardinalDirection, maxPortsPerSide) {
+  for(var j = 0; j < ports.length; j++) {
+    var posPercentage = 12 + 15 * (j % maxPortsPerSide);
```

Comment:
This is just putting the ports in y positions that work out to `15 * j + 12`? There are only five such positions in the range 0 to 100. And then, when you run out of space on one side, you'll just fill up the next. That's ten inputs and ten outputs per vertex, not infinitely many.

Perhaps they ought to be placed at positions that work out to `((j - 0.5) / ports.length) * 100`, so that they're guaranteed to be evenly spaced from each other and from the corners and each one gets a distinct position.